### ISSUE TITLE
GCC ICEs when TL is used in a template function

### ISSUE THREAD

**Issue by Quincunx271** [Quincunx271/TerseLambda#6]
Stripped down example: https://godbolt.org/z/4E6MerhfT

Upstream issue: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=101043

**Comment by Quincunx271** [comment]
The upstream issue doesn't seem to have any progress, and I took a quick look at the GCC code to see if it was something I could potentially hack a solution for, but I think the investment of understanding GCC is a bit high for now.

However, I played around with the reproducer, and I think we may be able to hack a solution into TerseLambda using lambdas in unevaluated contexts. That didn't use to be possible, because not many compilers supported it, but it's been a couple years since then, so more compilers do support that now. Even so, I could use the unevaluated context lambda for GCC only.

In short, I have a workaround in mind.

**Comment by Quincunx271** [comment]
Unfortunately, this workaround seems to run into another ICE:

```
/home/<...>/TerseLambda/tests/gcc-reproducer.test.cpp:14:9: internal compiler error: in finish_expr_stmt, at c
p/semantics.c:860
   14 |         std::erase_if(lhs, [&] TL(!rhs.contains(_1)));
      |         ^~~
```

**Comment by Quincunx271** [comment]
Filed: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=106398
